src/braket/pennylane_plugin/ahs_device.py

```diff
@@ -0,0 +1,575 @@
+    ):
+        if not shots:
+            raise RuntimeError(f"This device requires shots. Received shots={shots}")
+        # Simulator default of 0 not suitable for AHS simulator, use DEFAULT_SHOTS_QPU for both
+        elif shots == Shots.DEFAULT:
+            num_shots = AwsDevice.DEFAULT_SHOTS_QPU
+        else:
+            num_shots = shots
+
+        super().__init__(wires=wires, shots=num_shots)
+
+        self._device = device
+        self.register = None
+        self.pulses = None
+        self.ahs_program = None
+        self._task = None
+
+    def apply(self, operations: List[ParametrizedEvolution], **kwargs):
+        """Convert the pulse operation to an AHS program and run on the connected device
+
+        Args:
+            operations(List[ParametrizedEvolution]): a list containing a single
+                ParametrizedEvolution operator
+        """
+
+        if not np.all([op.name in self.operations for op in operations]):
+            raise NotImplementedError(
+                "Device {self.short_name} expected only operations "
+                "{self.operations} but recieved {operations}"
+            )
+        self._validate_operations(operations)
+        ev_op = operations[0]  # only one!
+
+        self._validate_pulses(ev_op.H.pulses)
+        ahs_program = self.create_ahs_program(ev_op)
+        self._task = self._run_task(ahs_program)
+
+    @property
+    def task(self):
+        return self._task
+
+    @property
+    def samples(self):
+        if self._task:
+            return self._task.result()
+        return None
+
+    def _run_task(self, ahs_program: AnalogHamiltonianSimulation):
+        """Run and return a task executing the AnalogueHamiltonianSimulation program on
+        the device"""
+        raise NotImplementedError("Running a task not implemented for the base class")
+
+    def _ahs_program_from_evolution(self, evolution: ParametrizedEvolution):
+        """Create AHS program for upload to hardware from a ParametrizedEvolution
+
+        Args:
+            evolution (ParametrizedEvolution): the PennyLane operator describing the pulse
+                to be converted into an Analogue Hamiltonian Simulation program
+
+        Returns:
+            AnalogHamiltonianSimulation: a program containing the register and drive
+                information for running an AHS task on simulation or hardware"""
+
+        # sets self.pulses to be the evaluated pulses (now only a function of time)
+        self._evaluate_pulses(evolution)
+        self._create_register(evolution.H.settings.register)
+
+        time_interval = evolution.t
+
+        # no gurarentee that global drive is index 0 once we start allowing more just global drive
+        drive = self._convert_pulse_to_driving_field(self.pulses[0], time_interval)
+
+        return AnalogHamiltonianSimulation(register=self.register, hamiltonian=drive)
+
+    def create_ahs_program(self, evolution: ParametrizedEvolution):
```

Comment:
Looking at the nature of this function, it probably belongs in translation.py (along with the accompanying private methods)

Reply:
I would prefer to leave this here, since it is device-specific (i.e. it is overwritten on the hardware device and requires access to the device to discretize). Additionally, it saves some of the information, like the register, on the device object during processing. (Again, this comes back to allowing users access to the objects created when going from their input to the hardware upload, so they can inspect and understand any differences.)

I've moved the device-independent, purely internal functions involved in translating from PennyLane to the `translations.py` file, but currently I've opted to leave it so this one is on the device, and have it set properties like `device.register` for users to access. Essentially functions translating HardwarePulse objects (rather than the initial operator) to Braket upload information are in `translation.py`. I'll push those changes soon!

Reply:
This adds quite a few lines of code to the translation file - about 200 in this PR, and probably another 200 in the local drive PR. Should I make a separate file that is `ahs_translation.py`? This will otherwise get quite long, especially if we also add translation functions for the transmon pulses in the future.

Reply:
I kept them in a separate `ahs_translation` file for now, let me know what you think

Reply:
You can even put them in the existing `translation.py` file.